onetl/connection/db_connection/kafka/kafka_ssl_protocol.py

```diff
@@ -0,0 +1,134 @@
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional
+
+from pydantic import Field, SecretStr, validator
+
+from onetl._util.file import is_file_readable
+from onetl.impl import GenericOptions, LocalPath
+
+if TYPE_CHECKING:
+    from onetl.connection import Kafka
+
+from onetl.connection.db_connection.kafka.kafka_protocol import KafkaProtocol
+
+
+class KafkaSSLProtocol(KafkaProtocol, GenericOptions):
+    """
+    Class encapsulates all the necessary configurations and interactions for managing
+    SSL protocols within a Kafka system. It serves as a handler for different keystore types and truststore configurations,
+    allowing the connection to Kafka brokers through SSL.
+
+    See:
+    * `IBM Documentation <https://www.ibm.com/docs/en/cloud-paks/cp-biz-automation/19.0.x?topic=fcee-kafka-using-ssl-kerberos-authentication>`_
+    * `Kafka Documentation <https://kafka.apache.org/090/documentation.html>`_
+    * `How to use PEM Certificates with Kafka <https://codingharbour.com/apache-kafka/using-pem-certificates-with-apache-kafka/>`_
+
+    Examples
+    --------
+
+    PEM certificates as raw strings:
+
+    .. code:: python
+
+        ssl = (
+            Kafka.SSLProtocol(
+                keystore_type="PEM",
+                keystore_certificate_chain="-----BEGIN CERTIFICATE-----MIIDZjC...-----END CERTIFICATE-----",
+                keystore_key="-----BEGIN ENCRYPTED PRIVATE KEY-----MIIDZjC..-----END ENCRYPTED PRIVATE KEY-----",
+                key_password="<private_key_password>",
+                truststore_type="PEM",
+                truststore_certificates="-----BEGIN CERTIFICATE-----MICC...-----END CERTIFICATE-----",
+            ),
+        )
+
+    PEM certificates as file paths:
+
+    .. code:: python
+
+        ssl = Kafka.SSLProtocol(
+            keystore_type="PEM",
+            keystore_location="/path/to/file/containing/certificate/chain.pem",
+            key_password="<private_key_password>",
+            truststore_type="PEM",
+            truststore_location="/path/to/truststore/certificate.pem",
+        )
+
+    JKS store path:
+
+    .. code:: python
+
+        ssl = Kafka.SSLProtocol(
+            keystore_type="JKS",
+            keystore_location="/opt/keystore.jks",
+            keystore_password="password",
+            truststore_type="JKS",
+            truststore_location="/opt/truststore.jks",
+            truststore_password="password",
+        )
+
+    Custom options:
+
+    .. code:: python
+
+        ssl = Kafka.SSLProtocol.parse(
+            {
+                "keystore_type": "JKS",
+                "keystore_location": "/opt/keystore.jks",
+                "keystore_password": "password",
```

Comment:
```suggestion
                "ssl.keystore.type": "JKS",
                "ssl.keystore.location": "/opt/keystore.jks",
                "ssl.keystore.password": "password",
```
Я тут подумал, такой вариант может быть более очевиден для пользователей, которые только знакомятся с инструментом, и использовали нейминг из других клиентов